nixos/modules/programs/slock.nix

```diff
@@ -16,16 +16,17 @@ in
           Whether to install slock screen locker with setuid wrapper.
         '';
       };
+      package = mkPackageOption pkgs "slock" {};
     };
   };
 
   config = mkIf cfg.enable {
-    environment.systemPackages = [ pkgs.slock ];
+    environment.systemPackages = [ cfg.package ];
     security.wrappers.slock =
       { setuid = true;
         owner = "root";
         group = "root";
-        source = "${pkgs.slock.out}/bin/slock";
+        source = lib.getExe cfg.package;
```

Comment:
Note that this will cause a bit warning. Either set `meta.mainProgram` or use `getExe' cfg.package "slock"`

Reply:
Slock does have `meta.mainProgram`, but only in unstable, it hasn't been backported to 23.11. Should we backport it (so that this commit could be backported as well) or just wait until the 24.05 release?

Reply:
Anyway, since backporting is a separate process, i don't think that we should care about it at the moment